landlab/ca/cfuncs.pyx

```diff
@@ -133,13 +123,13 @@ cdef class Event:
 @cython.boundscheck(True)
 @cython.wraparound(False)
 cdef int current_link_state(
-    DTYPE_INT_t link_id,
-    np.ndarray[DTYPE_INT_t, ndim=1] node_state,
-    np.ndarray[DTYPE_INT_t, ndim=1] node_at_link_tail,
-    np.ndarray[DTYPE_INT_t, ndim=1] node_at_link_head,
-    np.ndarray[DTYPE_INT8_t, ndim=1] link_orientation,
-    DTYPE_INT_t num_node_states,
-    DTYPE_INT_t num_node_states_sq,
+    const long link_id,
+    const id_t [:] node_state,
```

Comment:
For the most part, I've used fused data types to account for the way different operating systems represent integers. On Windows, for example, ints are sometimes represented as 64bit and sometimes as 32bit.

Using fused types shouldn't slow down the code very much, if at all, but they most definitely slow down compilation as *cython* generates a separate function for each combination of data types.

In the future we may want to remove some of these fused data types but this will depend on this Windows issue.